### PR TITLE
TFTRT: Support LeakyRelu op

### PR DESCRIPTION
**Pull request by trevor-m**
TRT adds support for a LeakyRelu activation in 5.1.

TF LeakyRelu Op is registered here: https://github.com/tensorflow/tensorflow/blob/cb96167443b3a5d30183e8b95f615fd29cc1e77a/tensorflow/core/ops/nn_ops.cc#L1011

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -2126,11 +2126,18 @@ TEST_F(OpConverterTest, ConvertActivation) {
         "Relu is only implemented for tensors, at my_act");
   }
 
+  const float ALPHA = 0.2f;
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: nit: kAlpha to follow the naming in the style guide. We should even mark this as constexpr.
https://google.github.io/styleguide/cppguide.html#Constant_Names

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -2126,11 +2126,18 @@ TEST_F(OpConverterTest, ConvertActivation) {
...
+
   // Get nodedef for activation layer.
-  auto get_act_nodedef = [](string op_name) -> NodeDef {
+  auto get_act_nodedef = [ALPHA](string op_name) -> NodeDef {
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: I believe that we don't need to capture constexpr or primitive const type variables.
See https://stackoverflow.com/a/49027502/9501553

**Review by smit-hinsu — DISMISSED** [review]
Thanks, Trevor! Looks great. Just two minor suggestions for the constant.

Sorry for the delay for the review.

**Comment by trevor-m** [comment]
Thanks @smit-hinsu !

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (smit-hinsu) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: s/kALPHA/kAlpha? (Full words are never capitalized. Even abbreviations use Camel case with only the first letter capitalized.)
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

**Comment by trevor-m** [comment]
Hey @smit-hinsu 

We found that TRT has performance issues with this LeakyReLU layer. We are going to change the conversion to use other ops instead for the time being.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: Whoops, thanks for correcting!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 3]
```diff
@@ -2430,6 +2430,69 @@ tensorflow::Status ConvertPool(OpConverterParams* params) {
...
+  if (alpha < 0.0f || alpha > 1.0f) {
+    return tensorflow::errors::Unimplemented(
+        "Alpha value for LeakyRelu must be between 0 and 1, at ",
+        node_def.name());
+  }
+  if (params->validation_only) return tensorflow::Status::OK();
+
+  // Input Tensor
+  const nvinfer1::ITensor* tensor = inputs.at(0).tensor();
+  // Create const for alpha. The constant has to have the same rank as the input
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: This pattern is duplicated in ConvertRelu6 and ConvertSquare as well. let's add a helper function?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 4]
```diff
@@ -2430,6 +2430,69 @@ tensorflow::Status ConvertPool(OpConverterParams* params) {
...
+  }
+  TRT_ShapedWeights weights = params->weight_store->GetTempWeights(
+      tensorflow::DataType::DT_FLOAT, dims);
+  auto weights_ptr =
+      static_cast<float*>(const_cast<void*>(weights.GetValues()));
+  weights_ptr[0] = alpha;
+  nvinfer1::ITensor* const_alpha_tensor =
+      params->converter->CreateConstantLayer(weights, dims);
+  TFTRT_RETURN_ERROR_IF_NULLPTR(const_alpha_tensor, node_def.name());
+  params->converter->ProvideQuantizationRange(const_alpha_tensor, -alpha,
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Not very familiar with the quantization in TensorRT but shouldn't the min and max both be alpha here? It's true that TensorRT only has symmetric ranges but shouldn't we specify the right values here given that we have min and max?

**Review by smit-hinsu — DISMISSED** [review]
Nice! Thanks for the quick update!

Curious about how the performance regression was detected?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 5]
```diff
@@ -351,6 +351,26 @@ nvinfer1::ITensor* Converter::CreateConstantLayer(
   return trt_tensor;
 }
 
+tensorflow::Status Converter::CreateBroadcastableScalarConstant(
```
Comment (smit-hinsu) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Passing params that has a pointer to Converter to some method in Converter does not sound very intuitive to me.

How about making this method a free helper method? We anyway does not seem to be using `this` pointer and directly using params->converter. 

Other option would be to not pass in params and instead pass node_name and weight_store but that will increase the number of arguments.

Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Yes that makes sense!

**Review by smit-hinsu — APPROVED** [review]
Thanks, Trevor!

**Pull request merged**